File: source/1.两数之和.cpp
/*
 * @lc app=leetcode.cn id=1 lang=cpp
 *
 * [1] 两数之和
 */
#include "cpp_includes.h"
// @lc code=start
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hash;
        for (int i = 0; i < nums.size(); i++) {
            int nums2 = target - nums[i];
            if (hash.count(nums2))
                return {hash[nums2], i};
            else
                hash[nums[i]] = i;
        }

        return {-1, -1};
    }
};
// @lc code=end

File: source/1.两数之和.py
#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#


# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict = {}
        for i, num in enumerate(nums):
            if target - num in dict:
                return [dict[target - num], i]
            dict[nums[i]] = i
        return []


# @lc code=end
